src/Exceptions/OrangeDamException.php

```diff
@@ -6,50 +6,52 @@
 use GuzzleHttp\Psr7\Response;
 use GuzzleHttp\Exception\RequestException;
 
-class OrangeDamException extends Exception {
-
-  /**
-   * Response instance.
-   *
-   * @var null|Response
-   */
-  protected $response;
-
-  /**
-   * Returns response instance.
-   *
-   * @return null|Response
-   */
-  public function getResponse() {
-    return $this->response;
-  }
-
-  /**
-   * Creates method.
-   */
-  public static function create(RequestException $guzzleException): self {
-    $e = new static(
-      static::sanitizeResponseMessage($guzzleException->getMessage()),
-      $guzzleException->getCode(),
-      $guzzleException
-    );
-
-    $e->response = $guzzleException->getResponse();
-
-    return $e;
-  }
-
-  /**
-   * Removes sensitive data from response message.
-   *
-   * @param string $message
-   *   Response message.
-   *
-   * @return string
-   *   Sanitized response message.
-   */
-  protected static function sanitizeResponseMessage(string $message): string {
-    return preg_replace('/(token)=[a-z0-9-]+/i', '$1=***', $message);
-  }
-
+class OrangeDamException extends Exception
+{
+    /**
+     * Response instance.
+     *
+     * @var null|Response
+     */
+    protected $response;
+
+    /**
+     * Returns response instance.
+     *
+     * @return null|Response
+     */
+    public function getResponse()
+    {
+        return $this->response;
+    }
+
+    /**
+     * Creates method.
+     */
+    public static function create(RequestException $guzzleException): self
+    {
+        $e = new static(
```

Comment:
<img width="704" alt="CleanShot 2023-06-21 at 21 40 04@2x" src="https://github.com/ChromaticHQ/orange-dam-php/assets/20355/d0b3a368-7062-40cc-87e6-f6144d4f93a7">

https://phpstan.org/blog/solving-phpstan-error-unsafe-usage-of-new-static

@adamzimmermann Can you take a look at this link and let me know how you think we can resolve this? We can't currently mark this class `final` as the other two exception types extend it, nor can we currently make it `abstract` as it is used directly (but we could change that). What do you think?

Reply:
I'm going to tag @ikorgik in here as he wrote this code and is much more familiar with it than I am. I can give this more attention later today if needed too.

Reply:
Hey @ikorgik! I have a possible workaround that involves declaring `final` on the constructor:

```php
final public function __construct(string $message = "", int $code = 0, ?Throwable $previous = null)
{
    parent::__construct($message, $code, $previous);
}
```

but before I commit that, I wanted to understand why we are using static() here 👇🏻  in the first place. Do we need to or could we avoid it completely?

Reply:
Hey Mark, I don't remember why we might need it. 
Most likely I took a part of code from my any other work and left this part as is since it works. 
If we have a way to improve it - let's do it.